### DISCUSSION TITLE
Carry-less multiplication

### DISCUSSION THREAD

**Discussion by kreuzerkrieg** [jfalcou/eve#1176] [Q&A]
Is carry-less multiplication supported by EVE? Something like PCLMULQDQ instruction?

**Comment by jfalcou** [comment]
We don't at the moment. What are your use case ?

**Comment by kreuzerkrieg** [comment]
SIMD accelerated CRC32 calculation following this [document](https://www.intel.com/content/dam/www/public/us/en/documents/white-papers/fast-crc-computation-generic-polynomials-pclmulqdq-paper.pdf
). I have intrinsic version of this code and would like to try and implement it using EVE

**Reply by jfalcou** [reply]
OK. At the moment, we don't support those. We can surely does, this means finding a way to implement it on non-X86 architectures.
Meanwhile, eve::wide can be directly passed to intrinsics so you can rewrite the rest fo the code without issue then just pass w/e to these intrinsics.

**Reply by DenisYaroshevskiy** [reply]
Posted an example below

**Reply by kreuzerkrieg** [reply]
> We can surely does, this means finding a way to implement it on non-X86 architectures.

https://en.wikipedia.org/wiki/CLMUL_instruction_set
Intel, AMD, ARMv8 and even SPARC

**Comment by DenisYaroshevskiy** [comment]
FYI: you can always mix eve with intrinsics if you need to https://godbolt.org/z/Mvbhj77ch

**Reply by kreuzerkrieg** [reply]
ha! looks interesting, will try to do that
